src/main/flight/mixer.c

```diff
@@ -729,17 +738,17 @@ static void applyMixToMotors(float motorMix[MAX_SUPPORTED_MOTORS])
     // Now add in the desired throttle, but keep in a range that doesn't clip adjusted
     // roll/pitch/yaw. This could move throttle down, but also up for those low throttle flips.
     for (int i = 0; i < motorCount; i++) {
-        float motorOutput = motorOutputMin + (motorOutputRange * (motorOutputMixSign * motorMix[i] + throttle * currentMixer[i].throttle));
+        float _throttle = throttle;
+        if (currentControlRateProfile->thrust_linearization_level && !currentControlRateProfile->throttle_linearization) {
+            // If throttle is not wanted to be linearized too, we apply a counter compensation
+            _throttle = scaleRangef(currentControlRateProfile->thrust_linearization_level, 0, 100, throttle, sqrtf(throttle));
+        }
+        float motorOutput = thrustToMotorOutput(motorOutputMixSign * motorMix[i] + _throttle * currentMixer[i].throttle);
         if (mixerIsTricopter()) {
             motorOutput += mixerTricopterMotorCorrection(i);
         }
         if (failsafeIsActive()) {
-            if (isMotorProtocolDshot()) {
-                motorOutput = (motorOutput < motorRangeMin) ? disarmMotorOutput : motorOutput; // Prevent getting into special reserved range
-            }
-            motorOutput = constrain(motorOutput, disarmMotorOutput, motorRangeMax);
-        } else {
-            motorOutput = constrain(motorOutput, motorRangeMin, motorRangeMax);
```

Comment:
I removed this because constrain is already done in thrustToMotorOutput function